### PR TITLE
Clean up some remaining dual reads for datasets. 

### PR DESCRIPTION
**Pull request by lostlevels**
Since all uploads are already migrated, this does not change the existing output of any function. As spotted by @Roukoswarf

### CODE REVIEW

File: data/store/mongo/mongo_data.go [hunk 1]
```diff
@@ -39,21 +39,12 @@ func (d *DataRepository) ListUserDataSets(ctx context.Context, userID string, fi
...
-		return dataSet, nil
-	}
-	return d.DatumRepository.GetDataSet(ctx, id)
+func (d *DataRepository) GetDataSet(ctx context.Context, uploadID string) (*data.DataSet, error) {
+	return d.DataSetRepository.GetDataSet(ctx, uploadID)
 }
 
-func (d *DataRepository) GetDataSetByID(ctx context.Context, dataSetID string) (*upload.Upload, error) {
-	return d.DataSetRepository.GetDataSetByID(ctx, dataSetID)
+func (d *DataRepository) GetDataSetByID(ctx context.Context, uploadID string) (*upload.Upload, error) {
```
Comment (darinkrauss) [inline data/store/mongo/mongo_data.go:46]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

File: data/store/mongo/mongo_data_set.go [hunk 2]
```diff
@@ -113,26 +114,26 @@ func (d *DataSetRepository) EnsureIndexes() error {
 	})
 }
 
-func (d *DataSetRepository) GetDataSetByID(ctx context.Context, dataSetID string) (*upload.Upload, error) {
+func (d *DataSetRepository) GetDataSetByID(ctx context.Context, uploadID string) (*upload.Upload, error) {
```
Comment (darinkrauss) [inline data/store/mongo/mongo_data_set.go:117]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

File: data/store/mongo/mongo_data.go [hunk 3]
```diff
@@ -39,21 +39,12 @@ func (d *DataRepository) ListUserDataSets(ctx context.Context, userID string, fi
...
-	// Can read only from the new deviceDataSets collection via DataSetRepository when migration completed.
-	dataSet, err := d.DataSetRepository.GetDataSet(ctx, id)
-	if err != nil {
-		return nil, err
-	}
-	if dataSet != nil {
-		return dataSet, nil
-	}
-	return d.DatumRepository.GetDataSet(ctx, id)
+func (d *DataRepository) GetDataSet(ctx context.Context, uploadID string) (*data.DataSet, error) {
```
Comment (darinkrauss) [inline data/store/mongo/mongo_data.go:42]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

File: data/store/mongo/mongo_data_set.go [hunk 4]
```diff
@@ -231,25 +232,25 @@ func (d *DataSetRepository) updateDataSet(ctx context.Context, id string, update
 	return d.GetDataSetByID(ctx, id)
 }
 
-func (d *DataSetRepository) GetDataSet(ctx context.Context, id string) (*data.DataSet, error) {
+func (d *DataSetRepository) GetDataSet(ctx context.Context, uploadID string) (*data.DataSet, error) {
```
Comment (darinkrauss) [inline data/store/mongo/mongo_data_set.go:235]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

File: data/store/store.go [hunk 5]
```diff
@@ -29,14 +29,14 @@ type DataSetRepository interface {
 	EnsureIndexes() error
 
 	GetDataSetsForUserByID(ctx context.Context, userID string, filter *Filter, pagination *page.Pagination) ([]*upload.Upload, error)
-	GetDataSetByID(ctx context.Context, dataSetID string) (*upload.Upload, error)
+	GetDataSetByID(ctx context.Context, uploadID string) (*upload.Upload, error)
```
Comment (darinkrauss) [inline data/store/store.go:32]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

File: data/store/store.go [hunk 6]
```diff
@@ -29,14 +29,14 @@ type DataSetRepository interface {
...
 	CreateDataSet(ctx context.Context, dataSet *upload.Upload) error
 	UpdateDataSet(ctx context.Context, id string, update *data.DataSetUpdate) (*upload.Upload, error)
 	DeleteDataSet(ctx context.Context, dataSet *upload.Upload) error
 	DestroyDataForUserByID(ctx context.Context, userID string) error
 
 	ListUserDataSets(ctx context.Context, userID string, filter *data.DataSetFilter, pagination *page.Pagination) (data.DataSets, error)
-	GetDataSet(ctx context.Context, id string) (*data.DataSet, error)
+	GetDataSet(ctx context.Context, uploadID string) (*data.DataSet, error)
```
Comment (darinkrauss) [inline data/store/store.go:39]: [nit] Please stick with `dataSetID` as `uploadID` is really an "old" term.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
LGTM, but I defer to @Roukoswarf for confirmation that all data sets are in the `DataSetRepository` and approval.

**Review by darinkrauss — APPROVED** [review]
LGTM!

**Review by Roukoswarf — APPROVED** [review]
Looks good

**Pull request merged**